### ISSUE TITLE
[Alerting] management evolution proposal

### ISSUE THREAD

**Issue by gcoeugnet** [librenms/librenms#3098]
Hi,

Testing alert management, we've noticed these : 
- An alert keeps appearing in the alerts widget when the problem, which cause the alert, is solved. It doesn't automatically close the created alert.
- When an alert is acknowledged, no new alert is created in database if the problem occurs again.

Would it be possible to have an alerting system which : 
- close automatically an existing alert if the related problem is solved  ?
- generate a new alert when a previous related alert was acknowledged, to be notified of a new problem which concern the same ressource ?

I can try to modify the code to make alerting works like this and make a pull request but I would like to know if this kind of alerting management would be acceptable for you.

Regards,
Guillaume.

**Comment by laf** [comment]
I don't experience any of that. Alerts are closed once the issue is resolved and if an alert is acknowledged and then is fixed and cleared, another alert will be generated.

Please ensure you are upto date on your install and run ./validate.php

**Comment by f0o** [comment]
Sadly need to agree with laf, not experiencing that either.
Alerts are closed when condition is resolved regardless of acknowledgment-states.
If acknowledged always change in state (worsen or betters) notifications are being restarted again.

**Comment by gcoeugnet** [comment]
Hi,

I'm glad to read your comments ! I'm discovering the product and last week I made several tests which lead me to create this post. Fortunatly for me, I made a screenshot showing that a generated alert was still appearing in the console even if the device was detected as being up again. 

I check the installation with the validate.php and change my Mysql configuration regarding the STRICT_TRANS_TABLES which was enabled. But I don't guess this explains the following...

Indeed, today I redo the same tests reading your comments and this time, the alert disappears... Maybe I was drunk all days of the last week... But, joined to this post, is the proof that I'm not totaly mad... The picture shows that the alert was still appearing in the console even if librenms detected the device as being available again...

![alerting](https://cloud.githubusercontent.com/assets/17470139/13390347/16803964-dece-11e5-94d0-4270832ade01.png)

Have a nice day,
Regards.
